### DISCUSSION TITLE
xMiner HiveOS Guide

### DISCUSSION THREAD

**Discussion by nathanccxv** [CyberChainXyz/go-cyberchain#8] [Mining]
This guide will help you set up xMiner on HiveOS to mine CyberChain (CCX).

> ⚠️ **BETA Testing**: xMiner HiveOS integration is currently in testing phase. Please be aware that:
> - Some issues might occur
> - Your feedback is valuable for improvements

## Add CCX Wallet in HiveOS

1. Go to "Wallets" page
2. Click "Add Wallet" button
3. Fill in the following:
   - Coin: Enter "OPEN-CCX" and click Create ”OPEN-CCX“
   - Address: Your CCX wallet address
   - Name: Enter a name for your wallet (e.g. "wallet-1")

![Screenshot from 2024-12-30 23-23-13](https://github.com/user-attachments/assets/0cabad5c-7753-4583-af91-da635bf9e1ad)

## Create Flight Sheet

1. Go to "Flight Sheets" page
2. Configure the following:
   - Coin: Select "OPEN-CCX"
   - Wallet: Select your CCX wallet  (e.g. "wallet-1")
   - Pool: Select "Configure in miner"
   - Miner: Select "Custom"

![Screenshot from 2024-12-30 23-27-03](https://github.com/user-attachments/assets/c32de4e9-058b-46e3-b486-625fb4ce9de3)

3. In the Custom configuration:
   - Miner name: `xMiner`
   - Installation URL: `https://github.com/CyberChainXyz/xMiner-hiveos/releases/download/v0.2.0/xMiner-v0.2.0.tar.gz`
   - Hash algorithm: `----`
   - Wallet and worker template: `%WAL%.%WORKER_NAME%`
   - Pool URL: `ws://ccx.coolpool.top:14003`

![Screenshot from 2024-12-30 23-16-54](https://github.com/user-attachments/assets/d41620d2-895a-40ac-94d0-aa0d19faa9b0)

## Support

If you encounter any issues or need help, please comment in this thread.

**Comment by StianNOR** [comment]
xMiner-v0.2.0.tar.gz                                        100%[========================================================================================================================================>]   4.64M  9.01MB/s    in 0.5s    

2024-12-30 22:45:58 (9.01 MB/s) - ‘xMiner-v0.2.0.tar.gz’ saved [4868776/4868776]

> Unpacking xMiner-v0.2.0.tar.gz
xMiner/
xMiner/h-manifest.conf
xMiner/h-config.sh
xMiner/h-stats.sh
xMiner/h-run.sh
xMiner/xMiner

No config found

custom exited (exitcode=1), waiting to cooldown a bit

Skipping miner log rotation due to execution time < 30sec
No config found

But the Config is there but its a txt file.

drwxr-xr-x 2 user  128    4096 Dec 30 22:47 .
drwxr-xr-x 7 user user    4096 Dec 30 22:45 ..
-rw-r--r-- 1 root root     100 Dec 30 22:45 config.txt
-rwxr-xr-x 1 user  128     869 Dec 30 22:45 h-config.sh
-rw-r--r-- 1 user  128     483 Dec 30 22:45 h-manifest.conf
-rwxr-xr-x 1 user  128     371 Dec 30 22:45 h-run.sh
-rwxr-xr-x 1 user  128    2313 Dec 30 22:45 h-stats.sh
-rwxr-xr-x 1 user  128 8825032 Dec 30 18:28 xMiner

**Reply by nathanccxv** [reply]
I've updated the xMiner HiveOS files, which should resolve the issue.

**Comment by Sudboyz** [comment]
Is there a way to solo mine to my own node with this?

**Reply by nathanccxv** [reply]
ref: https://github.com/CyberChainXyz/xMiner?tab=readme-ov-file#solo-mining

Run a [CyberChain](https://github.com/CyberChainXyz/go-cyberchain) node with mining feature enabled, set your mining pool to ws://node-ip:8546 in HiveOS.

**Comment by ptbeto** [comment]
![image](https://github.com/user-attachments/assets/72fa46f4-4ce7-4289-8ff1-8f77109c90d5)
The temps, hashrate, etc, does not appear in hiveos, is there a way to fix this?
And is there some tuning guide for the gpus or something?
Thanks.

**Reply by ptbeto** [reply]
![image](https://github.com/user-attachments/assets/0b7f7067-4738-4ac4-b14e-59009f82fd42)
Besides that, im getting a really high system load, is it normal?

**Reply by nathanccxv** [reply]
high system load 
https://github.com/CyberChainXyz/xMiner/issues/1

The temps, hashrate, etc, does not appear in hiveos
https://github.com/CyberChainXyz/xMiner-hiveos/issues/1

**Reply by nathanccxv** [reply]
> And is there some tuning guide for the gpus or something?

You can find some GPU overclocking info in the group at https://discord.com/invite/2K99ewZdjr for reference.

> The temps, hashrate, etc, does not appear in hiveos, is there a way to fix this?

Also, can you locate the /var/log/miner/xMiner/log file on your HiveOS machine? If so, could you please submit its contents at https://github.com/CyberChainXyz/xMiner-hiveos/issues/1?

**Comment by josean0102** [comment]
Hello, I am doing tests with mining, the wallet from which it is obtained, I have seen that exbitron has the currency, is this wallet valid. On the other hand, I have seen that they talk about metamask, could you help me?

**Reply by nathanccxv** [reply]
**Don't use exchange addresses!!** 

You can use any self-custody EVM wallet.

It's recommended to use MetaMask. Visit https://cyberchain.xyz/ to add CyberChain to MetaMask.
